### PR TITLE
✨ 800 - Add collapsable rule sidebar

### PR DESCRIPTION
**Pull request by Aibono1225**
Fixes https://github.com/SSWConsulting/SSW.Rules/issues/800

Changes I made:
- Added new sidebar to single rule view.
- Added Authors + Categories + Related Rules to the new sidebar
- Removed Authors + Categories + Related Rules from bottom section
- Make the sidebar responsive and sticky
- Added button to collapse the sidebar

Link of done video: https://youtu.be/wrXbO_0MI0Y

image

image

### CODE REVIEW

File: src/images/clock-regular.svg
Comment (bradystroud) [inline src/images/clock-regular.svg:1]: I think we can use FontAwesome for this icon

File: src/components/categories/categories.js
Comment (bradystroud) [inline src/components/categories/categories.js:1]: This filename could be a little more descriptive e.g. `sidebar-categories`

**Review by bradystroud — CHANGES_REQUESTED** [review]

**Review by jaydenalchin — CHANGES_REQUESTED** [review]
Fantastic work and nice done video! Please see my feedback below: 

1. Content in the sidebar should be downsized slightly for better scalability. E.g. text-sm, w-12 h-12 on author bubbles.
2. Separate the sidebar sections with a little more space between for readability. 
3. The bookmark / edit / github icons should have more space between them for clickability.
4. The 'Ratings' section needs to be updated with more spacing & center alignment to compensate for the other sections moving into the sidebar - see mockup below

image
**Figure: Ratings section**

**Comment by jaydenalchin** [comment]
> Done ⭐ See below:
@Aibono1225 That ratings section is looking good! 

1. Can you please show the rule title & icons as well? 

I also think we should reevaluate the need for collapse/expand interaction you showed in the demo video. I feel it's adding clutter to the UI without providing much actual value. Would love to hear thoughts from @bradystroud @tiagov8

**Comment by Aibono1225** [comment]
> > Done ⭐ See below:
> > @Aibono1225 That ratings section is looking good!
> 
> 1. Can you please show the rule title & icons as well?

Done - here the screenshots:
destop
**Desktop View**

mobile
**Mobile View**



The rating section is also responsive:
772F1F38-F998-46A8-8CE7-AD18897B52AB

**Comment by Aibono1225** [comment]
As per Tiago's emails, the following changes have been made:

1. These old icons have been added back:  bookmark, edit(Netlify), and edit(GitHub) and 'see history' icon
2. The authors' section will now only display photos.
3. The authors'section and categories section now have 'info' links that show "how to add an author/category"
4. The related rules are no longer numbered.

Here is the new Done Videor: https://youtu.be/1ort1enSMLo

image
**Desktop View**


image
**Mobile View**

**Comment by tiagov8** [comment]
LGTM ✅

Thanks @Aibono1225 and @jaydenalchin for applying less is more ;)

**Comment by jaydenalchin** [comment]
Excellent work @Aibono1225 LGTM and thanks I know that was a lot of feedback to go through! ✨💖

I have one last suggestion that is not need-to-have but nice-to-have if @bradystroud or @tiagov8 agree:

1. For a nice subtle indication that there's overflow content, we could add a small fade-out effect on the top and bottom of the scrolling sidebar like this: https://pqina.nl/blog/fade-out-overflow-using-css-mask-image/

**Comment by Aibono1225** [comment]
Thank you @jaydenalchin . The link you provided was very helpful, and the new CSS effect looks so cool to me! 😊

I have added the fade-out effect to the sidebar, as you can see below:
image

**Review by jaydenalchin — APPROVED** [review]
Awesome work love this big update! 🎉🚀

File: src/style.css [hunk 1]
```diff
@@ -1808,7 +1790,8 @@ div.bad-example {
     padding: 1rem;
   }
   .reaction-grid {
-    grid-template-columns: auto 2rem 0.5rem 1rem 2.2rem 0.5rem 1rem 2.2rem 0.5rem 1rem 2.3rem 1rem auto;
+    /* grid-template-columns: auto 2rem 0.5rem 1rem 2.2rem 0.5rem 1rem 2.2rem 0.5rem 1rem 2.3rem 1rem auto; */
```
Comment (bradystroud) [inline src/style.css:1793]: can we remove this?

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: src/style.css (hunk 1 above)
Comment (Aibono1225) [reply src/style.css]: Done (●'◡'●)

**Review by bradystroud — APPROVED** [review]

**Pull request merged**